pkg/storkctl/schedulepolicy.go

```diff
@@ -60,6 +67,215 @@ func newGetSchedulePolicyCommand(cmdFactory Factory, ioStreams genericclioptions
+				dailyPolicy.ForceFullSnapshotDay = dailyForceFullSnapshotDay
+				if retain == 0 {
+					dailyPolicy.Retain = storkv1.DefaultDailyPolicyRetain
+				} else {
+					dailyPolicy.Retain = storkv1.Retain(retain)
+				}
+				err := dailyPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Daily = &dailyPolicy
+			case "weekly":
+				var weeklyPolicy storkv1.WeeklyPolicy
+				weeklyPolicy.Time = time
+				weeklyPolicy.Day = dayOfWeek
+				if retain == 0 {
+					weeklyPolicy.Retain = storkv1.DefaultWeeklyPolicyRetain
+				} else {
+					weeklyPolicy.Retain = storkv1.Retain(retain)
+				}
+				err := weeklyPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Weekly = &weeklyPolicy
+			case "monthly":
+				var monthlyPolicy storkv1.MonthlyPolicy
+				monthlyPolicy.Time = time
+				monthlyPolicy.Date = dateOfMonth
+				if retain == 0 {
+					monthlyPolicy.Retain = storkv1.DefaultMonthlyPolicyRetain
+				} else {
+					monthlyPolicy.Retain = storkv1.Retain(retain)
+				}
+				err := monthlyPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Monthly = &monthlyPolicy
+			default:
+				util.CheckErr(fmt.Errorf("need to provide a valid schedule policy type. Valid schedule types are Interval, Daily, Weekly and Monthly"))
+				return
+			}
+			var schedulePolicy storkv1.SchedulePolicy
+			schedulePolicy.Name = schedulePolicyName
+			schedulePolicy.Policy = policyItem
+			_, err := storkops.Instance().CreateSchedulePolicy(&schedulePolicy)
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+			msg := fmt.Sprintf("schedule policy %v created successfully", schedulePolicy.Name)
+			printMsg(msg, ioStreams.Out)
+		},
+	}
+	// Picking up user inputs and setting the defaults for flags
+	createSchedulePolicyCommand.Flags().StringVarP(&schedulePolicyType, "policy-type", "t", "Interval", "Select Type of schedule policy to apply. Interval / Daily / Weekly / Monthly.")
+	createSchedulePolicyCommand.Flags().IntVarP(&intervalMinutes, "interval-minutes", "i", 30, "Specify the interval, in minutes, after which Portworx should trigger the operation.")
+	createSchedulePolicyCommand.Flags().IntVarP(&retain, "retain", "", 0, "For backup operations, specify how many backups Portworx should retain.")
+	createSchedulePolicyCommand.Flags().StringVarP(&time, "time", "", "12:00AM", "Specify the time of the day in the 12 hour AM/PM format, when Portworx should trigger the operation.")
+	createSchedulePolicyCommand.Flags().StringVarP(&dailyForceFullSnapshotDay, "force-full-snapshot-day", "", "MONDAY", "For daily scheduled backup operations, specify on which day to trigger a full backup.")
+	createSchedulePolicyCommand.Flags().StringVarP(&dayOfWeek, "day-of-week", "", "Sunday", "Specify the day of the week when Portworx should trigger the operation. You can use both the abbreviated or the full name of the day of the week.")
```

Comment:
Specify the day of the week when Stork should trigger the operation. You can use both the abbreviated or the full name of the day of the week.